### PR TITLE
rambox-pro: fix missing icons, autostart and clean up dir layout

### PR DESCRIPTION
**Pull request by cawilliamson**
###### Motivation for this change
The main motivation for this change was to add actual icons to Rambox Pro since the lack of any icons was bugging me and no doubt any other users of the package.

I have also resolved an issue where the Electron application itself was autostarted at login instead of Rambox Pro like it should be.

I also decided to clean up the directory structure in-line with other similar packages.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/rambox/pro.nix [hunk 1]
```diff
@@ -16,22 +16,28 @@ stdenv.mkDerivation rec {
...
       --prefix PATH : ${xdg_utils}/bin
   '';
 
   desktopItem = makeDesktopItem {
     name = "rambox-pro";
     exec = "ramboxpro";
+    icon = "ramboxpro";
     type = "Application";
     desktopName = "Rambox Pro";
+    caterogies = "Network;";
```
Comment (pbogdan) [inline pkgs/applications/networking/instant-messengers/rambox/pro.nix]: The attribute name appears to be misspelt.

Comment (cawilliamson) [reply pkgs/applications/networking/instant-messengers/rambox/pro.nix]: Yep, my bad. :(

**Comment by cawilliamson** [comment]
Once this changeset is merged - I'll push an update to v1.2.1.

**Comment by gnidorah** [comment]
cc @cawilliamson

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 75269` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- rambox-pro
</details>

**Pull request merged**